CARDS-1306/CARDS-1353: Add support for sections that stick to the bottom (footer section) of the page when scrolling

marta- opened the pull request:
To test, use Footer Sections Test questionnaire available in the test run mode.

marta- committed:
CARDS-1306/CARDS-1353: Add support for sections that stick to the bottom (footer section) of the page when scrolling

sashaandjic commented:
no more header setting?

marta- commented:
I split 1306 into 2 subtasks, one for header and one for footer. This PR is for subtask 1353 - footer.

As I commented in the original PR #724 which is now closed, support for headers was buggy and will be reimplemented in a separate PR. There's no reason to hold merging the support for footers because of the buggy headers.

sashaandjic commented:
Preview in designer does not resemble the look of the form. In the Radio form, I marked the last section to stick to the bottom. When I preview I only see this:
![image](https://user-images.githubusercontent.com/23243244/129834028-3b8f6334-5425-48a6-94c4-561f64bd3de4.png)
Meanwhile the actual form looks like this:
https://user-images.githubusercontent.com/23243244/129834313-3e40f5f6-e223-498b-a84e-93a80d9cadfb.mp4

sashaandjic reviewed (approved):
tested

sdumitriu commented on test-resources/src/main/resources/SLING-INF/content/Questionnaires/StickySectionTest.xml:
```diff
@@ -0,0 +1,364 @@
+<!--
```
Please rename the file, it's only about footers now. Or do you plan to add sticky headers to it as well once the other PR is ready to merge?

marta- replied:
Yes, header sections are added to this questionnaire in PR #740.

sdumitriu reviewed (commented):
Just like with headers, footer sections have an overlap issue with pagination controls. Turning the pagination on for the footers questionnaire will make the footer question hidden until the bottom of the page is reached. Not a blocker, though, we could merge as is and add support for that use case later.

marta- commented:
> Just like with headers, footer sections have an overlap issue with pagination controls. Turning the pagination on for the footers questionnaire will make the footer question hidden until the bottom of the page is reached. Not a blocker, though, we could merge as is and add support for that use case later.

Right, forgot to account for pagination. It can easily be implemented in a similar way as the header sections stay under the resource header in PR #740. Please review #740, and if all looks good I can add similar support for pagination.

sashaandjic commented:
@Marta - please check this video. Look what is happening at the top.
![image](https://user-images.githubusercontent.com/23243244/129970465-c7186843-aa23-4e0c-9838-414884869a1b.png)

https://user-images.githubusercontent.com/23243244/129970539-2f2f51fd-dc46-43e0-8b78-3172b792e6cb.mp4

marta- commented:
> > Just like with headers, footer sections have an overlap issue with pagination controls. Turning the pagination on for the footers questionnaire will make the footer question hidden until the bottom of the page is reached. Not a blocker, though, we could merge as is and add support for that use case later.
> 
> Right, forgot to account for pagination. It can easily be implemented in a similar way as the header sections stay under the resource header in PR #740. Please review #740, and if all looks good I can add similar support for pagination.

Added the fix for footer sections on paginated forms to PR #740 (commit bd224fac) because it relies on some changes that are already in that branch.

marta- commented:
> @marta - please check this video. Look what is happening at the top.

@sashaandjic I can't find the right browser window height to replicate this, but thanks to the video I think I know what's going on and I'm hoping commit 088917c takes care of it.

marta- committed:
CARDS-1306/CARDS-1353: Add support for sections that stick to the bottom (footer section) of the page when scrolling
Addressing review comments: prevent the form header from being covered by a sticky footer section

sashaandjic commented:
I made the video with the same setup. Chrome zoom 75%.
Notice how when landing on the paginated page with footer setting the top and the bottom of the page is hidden

https://user-images.githubusercontent.com/23243244/129995904-7971da21-722d-4331-a412-58bda3013b18.mp4

marta- commented:
> I made the video with the same setup. Chrome zoom 75%.
> Notice how when landing on the paginated page with footer setting the top and the bottom of the page is hidden

The bottom issue is the same as what @sdumitriu reported, and I pushed a fix for it to the #740 PR instead of this one because the fix needed some code that was already in #740. Please test it there.

The top issue i would say it's not a priority, and not worth holding the PR for. Realistically, large sections shouldn't be set as footers.
